lib/Dialect/RTL/RTLFolds.cpp

```diff
@@ -706,3 +714,259 @@ void ConcatOp::getCanonicalizationPatterns(OwningRewritePatternList &results,
   };
   results.insert<Folder>(context);
 }
+
+//===----------------------------------------------------------------------===//
+// MuxOp
+//===----------------------------------------------------------------------===//
+
+OpFoldResult MuxOp::fold(ArrayRef<Attribute> constants) {
+  // mux (a, b, b) -> b
+  if (trueValue() == falseValue())
+    return trueValue();
+
+  // mux(0, a, b) -> b
+  // mux(1, a, b) -> a
+  auto pred = constants[0].dyn_cast_or_null<IntegerAttr>();
```

Comment:
Super nit but I'd use:
`if (auto pred = constants[0].dyn_cast_or_null<IntegerAttr>()) {`

to reduce the scope of pred.